nix/shell.go

```diff
@@ -286,6 +290,34 @@ func (s *Shell) writeDevboxShellrc() (path string, err error) {
 	return path, nil
 }
 
+func (s *Shell) copyShellSettings(shellSettingsDir string) {
+
+	// For now, we only need to do this for zsh shell
+	if s.name == shZsh {
+		// support zsh4humans by copying these files over
+		// https://github.com/romkatv/zsh4humans/blob/master/tips.md#backup-and-restore
+
+		// TODO savil: make this a glob search `.p10k*.zsh`
+		filenames := []string{".zshenv", ".p10k.zsh"}
+		for _, filename := range filenames {
+			fileOld := filepath.Join(filepath.Dir(s.userShellrcPath), filename)
+			if _, err := os.Stat(fileOld); errors.Is(err, fs.ErrExist) {
+				// this file may not be relevant for the user's setup.
+				continue
+			}
+
+			fileNew := filepath.Join(shellSettingsDir, filename)
```

Comment:
Can we somehow solve this in a more generic way, or in a way that the users can configure this themselves? Something along the lines of exporting an environment variable for devbox shellrc dir and have the users copy any files over?

Reply:
Darn. Moving `.p10k.zsh` to be linked from the shell-init-script doesn't seem to work. 

I think the reason is:
- the devbox shell's `.zshrc` first runs the user's `.zshrc` and afterwards runs the shell init hooks.
- the user's `.zshrc` has init logic for zsh4humans, but at this point the `.p10k.zsh` is missing and so the zsh4humans-wizard starts.

Reply:
okay, an ugly workaround for now is to hard-link the `.p10k.zsh` files from within the `.zshrc` prior to calling `z4h init`.

Reply:
for example, I did:
```
# within my .zshrc file:
...
if (( ${+ZDOTDIR} )); then
  ln ~/.p10k*.zsh $ZDOTDIR
fi


# Install or update core components (fzf, zsh-autosuggestions, etc.) and
# initialize Zsh. After this point console I/O is unavailable until Zsh
# is fully initialized. Everything that requires user interaction or can
# perform network I/O must be done above. Everything else is best done below.
z4h init || return
```

Reply:
oh well